src/k2/module/k23si/Module.cpp

```diff
@@ -233,12 +242,12 @@ bool K23SIPartitionModule::_isScanDone(const IndexerIterator& it, const dto::K23
 
 // Helper for handleQuery. Returns continuation token (aka response.nextToScan)
 dto::Key K23SIPartitionModule::_getContinuationToken(const IndexerIterator& it,
-                    const dto::K23SIQueryRequest& request, size_t response_size) {
+                    dto::K23SIQueryRequest& request, size_t response_size) {
```

Comment:
I think it is better to have this function take the dto::K23SIQueryResponse and fill in the token and flag, rather than change the flag in the request.